interfaces/apparmor/backend.go

```diff
@@ -548,6 +571,24 @@ func (b *Backend) Remove(snapName string) error {
 	return errUnload
 }
 
+func (b *Backend) RemoveLate(snapName string, rev snap.Revision, typ snap.Type) error {
+	logger.Noticef("remove late for snap %v (%s) type %v", snapName, rev, typ)
+	if typ != snap.TypeSnapd {
+		// late remove is relevant only for snap confine profiles
+		return nil
+	}
+
+	globs := []string{snapConfineProfileName(snapName, rev)}
+	_, removed, errEnsure := osutil.EnsureDirStateGlobs(dirs.SnapAppArmorDir, globs, nil)
+	// XXX: unloadProfiles() does not unload profiles from the kernel, but
+	// only removes profiles from the cache
+	errUnload := unloadProfiles(removed, apparmor_sandbox.CacheDir)
+	if errEnsure != nil {
+		return fmt.Errorf("cannot remove security profiles for snap %q (%s): %s", snapName, rev, errEnsure)
+	}
+	return errUnload
```

Comment:
(nitpick) This is slightly unusual that we handle errEnsure so late, maybe a small comment why we do it this way? Also if errUnload is set and errEnsure as well we silently drop errUnload (it never surfaces in a log or anything). Maybe fine but if so a small comment would be great.